gluten-core/src/main/scala/io/glutenproject/utils/InputPartitionsUtil.scala

```diff
@@ -29,24 +29,12 @@ case class InputPartitionsUtil(
     relation: HadoopFsRelation,
     selectedPartitions: Array[PartitionDirectory],
     output: Seq[Attribute],
+    bucketedScan: Boolean,
     optionalBucketSet: Option[BitSet],
     optionalNumCoalescedBuckets: Option[Int],
     disableBucketedScan: Boolean)
   extends Logging {
 
-  private val bucketedScan: Boolean = {
-    if (
-      relation.sparkSession.sessionState.conf.bucketingEnabled && relation.bucketSpec.isDefined
-      && !disableBucketedScan
-    ) {
-      val spec = relation.bucketSpec.get
-      val bucketColumns = spec.bucketColumnNames.flatMap(n => toAttribute(n))
-      bucketColumns.size == spec.bucketColumnNames.size
-    } else {
-      false
-    }
-  }
```

Comment:
@WangGuangxin  please help to review, thanks.

Reply:
LGTM